lib/metanorma/yaml2xml_collection.rb

```diff
@@ -0,0 +1,139 @@
+require "relaton-cli"
+# require "yaml"
+# require "tempfile"
+# require "relaton"
+# require "metanorma-cli"
+# require "nokogiri"
+
+module Metanorma
+  class Yaml2XmlColection
+    # @param file [String] YAML collection file path
+    def initialize(file)
+      @collection = YAML.load_file(file)
+      @docs = []
+      require "metanorma-#{doctype}"
+    end
+
+    # @param file [String] filename of YAML collection
+    # return [String] XML collection
+    def self.convert(file)
+      new(file).output
+    end
+
+    # @return [String] XML collection
+    def output
+      <<~END
+        <metanorma-collection xmlns="http://metanorma.org">
+          #{collection_bibdata}
+          #{manifest(@collection["manifest"])}
+          #{prefatory}
+          #{doccontainer}
+          #{final}
+        </metanorma-collection>
+      END
+    end
+
+    private
+
+    # @return [String] bibdata XML
+    def collection_bibdata
+      return unless @collection["bibdata"] 
+
+      Tempfile.open(["bibdata", ".yml"], encoding: "utf-8") do |f|
+        f.write(YAML.dump(@collection["bibdata"])) 
```

Comment:
What does this do?

Reply:
The code is written by Nick. I don't think we need to use a temp file.

Reply:
I think it is used to dump Collection Metadata. Again there should be a better location to handle Metadata.

Reply:
The code `YAML.dump(@collection["bibdata"])` does the same as `@collection["bibdata"].to_yaml`.
So it saves bibdata from collection to temp YAML file then converts the file to XML format using `Relaton::Cli::YAMLConvertor`. Since `Relaton::Cli::YAMLConvertor` accepts file name as an argument, we need to save the bibdata to YAML file before converting it.

Reply:
I.e. the code stores intermediate data to a temp file because the `Relaton::Cli::YAMLConvertor` accept file name as argument. We cant pass the data direct to the converter. To be able to do it we need to refactor relaton-cli.

Reply:
> To be able to do it we need to refactor relaton-cli

Do you mean using Relaton as a library for access? As you know, the parsing code in Relaton::Cli should really belong in Relaton proper... If so, yes a refactor will be necessary.